src/messages/MOSDMap.h

```diff
@@ -82,8 +82,24 @@ class MOSDMap final : public Message {
     using ceph::decode;
     auto p = payload.cbegin();
     decode(fsid, p);
+
     decode(incremental_maps, p);
+    for (auto iter = incremental_maps.begin();
+         iter != incremental_maps.end(); ++iter) {
+      bufferlist& bl = iter->second;
+      if (!bl.is_page_aligned()) {
```

Comment:
Calling rebuild_page_aligned() on an aligned bufferlist appears to be a noop and about as expensive as calling is_page_aligned().  For an unaligned bufferlist, calling is_page_aligned() first is simply extra cost.  I think removing the is_page_aligned() guards will be both more performant and clearer.